otherlibs/unix/unix.mli

```diff
@@ -212,6 +212,13 @@ val execv : string -> string array -> 'a
    the arguments [args], and the current process environment.
    These [execv*] functions never return: on success, the current
    program is replaced by the new one.
+
+   On Windows: the CRT simply spawns a new process and exits the
+   current one. This will have unwanted consequences if e.g.
+   another process is waiting on the current one.
```

Comment:
The StackOverflow answer you reference in #11754 makes an explicit mention of shell script usage, which sounds likely to be the way most users would observe this issue. It was not immediately clear to me reading this that the "another process is waiting on the current one" will have consequences in a shell (I thought of multi-process programming patterns at the level of the program itself). Maybe your word of warning could include an explicitly explanation of the shell stuff, for example to make it easier for people bitten by the issue to locate this explanation?

(I would even include your example where `echo before\n <program>\n echo after` turns into `before\n after\n <program output>`. It's visual.)

Reply:
Note that this is a pretty different semantics between unices and windows. The `Unix` library, despite its name, is a compatibility layer. 

Wouldn't we be better off emulating the `unix` semantics on Windows ? For example [here is what I do](https://github.com/b0-system/b0/blob/152c47864e9da478027bfa1113b892334abecffc/src/std/b0_std.ml#L3936-L3960) in one of my system helper library.

Reply:
@gasche Note that the term "console applications" (what is mentioned in the [SO answer](https://stackoverflow.com/questions/53218984/how-can-i-avoid-the-windows-exec-and-spawn-functions-splitting-arguments-at#comment93349909_53218984)) goes beyond shell scripts under Windows. I suspect he semantics will be incorrect for all GUI-less applications -- not just scripts. In the issue, I described the way we encountered the problem, but I suspect it would have been the same if we had replaced the 'echo' in our testing by another process. (I did not check though.) For me, the biggest problem is that any application waiting (in the `Unix.waitpid` sense) will be re-awakened before the `exec`'ed process has finished. This is almost never what you want. But I can certainly expand the description with more "syndromes".

@dbuenzli I partially agree. I too would like the Windows implementations of `Unix` to behave as much as their POSIX namesakes. However, I think your own implementation is _better_, but still not similar enough. In particular, the parent process is not replaced, it keeps executing. (Keeping alive probably memory, resources, etc) This is why I would prefer to document the current quirky behavior. (And also, I don't feel confident enough in my system programming skills to replace the current implementation.)

Reply:
@yakobowski I'm not disputing that my emulation might end up being surprising in certain scenarios (e.g. if you `excecv` for spawning a long running process) – it was also likely little tested in practice. 

But OTOH the current behaviour is a trap in my opinion -- for one thing is does the wrong thing w.r.t. controlling terminals (the `exec`ed cmd is like if you `cmd &`) . I think it would be better to either try to emulate more of the semantics and document the caveats or simply return ENOSYS on Windows.

In any case this is not blocker for merging this doc bits. But rather a discussion for #11892.